### DISCUSSION TITLE
registration self-service flow with native client (api) redirects to browser ui /login on existing account

### DISCUSSION THREAD

**Discussion by openscript** [ory/kratos#3409] [Q&A] [answered]
We have a native client (React Native) as well as a browser client for the self-service flows. When users register themself via a social login provider (OIDC) on our native app and there is already a user with this identifier, they will be redirected to the login self-service in the browser. We have set the `return_to` parameter in the native app.

How can we configure kratos, that when a user registers via the native app, coming back from the social provider and this identifier is already used they are redirected, the native login ui (for example via deeplinking) is shown?

**Comment by vinckr** [comment]
Hello @openscript 
To configure Ory Kratos for redirecting users to the native login UI in case of an identifier conflict during registration via a social provider, you need to follow the native app authentication flow as described in the [Ory Kratos documentation](https://www.ory.sh/docs/kratos/social-signin/native-apps).
Here are the steps you need to follow:
1. Initialize a login or registration flow in your native application through the Ory Network APIs. Set the return_to parameter to the URL of your native application. This URL will be used to redirect the user back to the app after the social sign-in. Also, set the return_session_token_exchange_code parameter to true to receive the session token exchange code in the response.
   ```
   // src/components/Routes/Login.tsx
   // Initialize the login or registration flow here
   ```
1. Upon rendering the form, let the user select the specific social sign-in provider. The native application then submits the form.
1. After the user has performed the social sign-in, they will be redirected back to the native application via an iOS Universal Link or Android App Link. The native application then exchanges the session token exchange code for a session token.
1. As part of this flow, Ory will redirect the browser to a callback. In this demo, the redirect URL is http://localhost:19006/Callback. Ensure that this redirect URL is allowed in your Ory configuration.
1. If the user identifier is already in use, you should handle this case in your application logic and redirect the user to the native login UI. You can use deep linking for this purpose.

Please note that the exact implementation may vary depending on your application's specific requirements and the social sign-in provider you are using.

**Reply by openscript** [reply]
Hello @vinckr 

Thank you for taking the time and give me such a detailed answer. I still believe it's a bug, that the `return_to` is not taking into account when an error happens. I can reproduce this on Ory Network. Currently Ory Network with Native clients is broken, so it is not usable for us:

https://github.com/ory/kratos/issues/3420
https://github.com/ory/kratos/issues/3417

I'm happy to collaborate to move forward on this issues!

**Comment by kmherrmann** [comment, marked as answer]
Thanks for filing this. We'll investigate and fix as needed.
